lib/SILOptimizer/LoopTransforms/ArrayBoundsCheckOpts.cpp

```diff
@@ -1200,6 +1232,15 @@ static bool hoistBoundsChecks(SILLoop *Loop, DominanceInfo *DT, SILLoopInfo *LI,
               Changed = true;
               continue;
             }
+            if (isComparisonKnownFalse(Builtin, *IV)) {
+              if (!FalseVal) {
+                FalseVal = SILValue(B.createIntegerLiteral(
```

Comment:
Are you sure you need the SILValue(...) here? I think just assigning the instruction should just work.